apps/x509.c

```diff
@@ -489,8 +489,12 @@ int x509_main(int argc, char **argv)
         goto end;
     }
 
+    if (newcert && infile != NULL) {
+        BIO_printf(bio_err, "ignoring -in option since -newcert is set\n");
+        infile = NULL;
+    }
     if (newcert && fkeyfile == NULL) {
-        BIO_printf(bio_err, "We need a public key, use -force_pubkey or -in\n");
+        BIO_printf(bio_err, "We need a public key, use -force_pubkey\n");
```

Comment:
'We need a public key" is too cryptic.  Be more explicit: "The -new option requires that a public key be set via -force_pubkey" or similar.

Reply:
I've changed this to:

> The -new option requires a public key to be set using -force_pubkey